src/disk.py

```diff
@@ -107,4 +108,58 @@ def disk_info(name, **kwargs):
     if not disk:
         return f"Unknown disk with name {name}" if human_readable else None
 
-    return _disk_infos(name, disk.props, **kwargs)
+    return _format_infos(name, disk.props, **kwargs)
+
+
+@dataclasses.dataclass
+class Health:
+    uptime: int | None
+    temperature: int | None
+    smartStatus: DiskState
+    smartSelftest: str
+    smartSelftestUpdated: datetime | int | None
+    smartAttributes: dict[str, str] | None = None
+
+    def asdict(self):
+        return dataclasses.asdict(self)
+
+
+def disk_health(name, **kwargs):
+    bus = sd_bus_open_system()
+    disk = Udisks2Manager(bus).get_disks().get(name)
+
+    human_readable = kwargs.get("human_readable", False)
+
+    if not disk or not not disk.props.get("smart_supported"):
+        # If disk does not exist or SMART is not supported by disk
+        return None
+
+    match disk.props:
+        case {"smart_power_on_seconds": uptime}:
+            # ATA
+            uptime = uptime * 3600
+        case {"smart_power_on_hours": uptime}:
+            # NVMe
+            pass
+        case _:
+            return None
+
+    smart_attributes = None
+    if issubclass(disk.iface, SmartMixin):
+        smart_attributes = disk.dbus_obj.get_smart_attributes()
+
+    temperature = disk.props["smart_temperature"]
+    smart_updated = disk.props["smart_updated"]
+    if not smart_updated:
+        smart_updated = None
+    if human_readable:
+        smart_updated = datetime.fromtimestamp(smart_updated) if smart_updated else "Unknown"
+
+    return Health(
+        uptime=uptime,
+        temperature=temperature,
+        smartStatus=DiskState.parse(disk.props),
+        smartSelftest=disk.props["smart_selftest_status"],
```

Comment:
The result here is a result code [here for NVMe](https://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.NVMe.Controller.html#gdbus-property-org-freedesktop-UDisks2-NVMe-Controller.SmartSelftestStatus) and [here for ATA](https://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Drive.Ata.html#gdbus-property-org-freedesktop-UDisks2-Drive-Ata.SmartSelftestStatus). I don't really know how to tackle this. Should it be translated on backend or on front-end?